### DISCUSSION TITLE
FieldInversion in DAfoam

### DISCUSSION THREAD

**Discussion by lnhang0618** [mdolab/dafoam#433] [General]
Dear Prof. He,
Can I run FieldInversion on v3 or later ?There is only runScript_v2.py in the tutorial.

**Comment by lnhang0618** [comment]
Traceback (most recent call last):
  File "runScript_v2.py", line 135, in <module>
    DASolver.addFamilyGroup(DASolver.getOption("designSurfaceFamily"), DASolver.getOption("designSurfaces"))
  File "/home/lnhang/dafoam/packages/miniconda3/lib/python3.8/site-packages/dafoam/pyDAFoam.py", line 3743, in getOption
    raise Error("%s is not a valid option name." % name)
dafoam.pyDAFoam.Error
Traceback (most recent call last):
  File "runScript_v2.py", line 135, in <module>
    DASolver.addFamilyGroup(DASolver.getOption("designSurfaceFamily"), DASolver.getOption("designSurfaces"))
  File "/home/lnhang/dafoam/packages/miniconda3/lib/python3.8/site-packages/dafoam/pyDAFoam.py", line 3743, in getOption
    raise Error("%s is not a valid option name." % name)
dafoam.pyDAFoam.Error
Traceback (most recent call last):
  File "runScript_v2.py", line 135, in <module>
    DASolver.addFamilyGroup(DASolver.getOption("designSurfaceFamily"), DASolver.getOption("designSurfaces"))
  File "/home/lnhang/dafoam/packages/miniconda3/lib/python3.8/site-packages/dafoam/pyDAFoam.py", line 3743, in getOption
    raise Error("%s is not a valid option name." % name)
dafoam.pyDAFoam.Error
Traceback (most recent call last):
  File "runScript_v2.py", line 135, in <module>
    DASolver.addFamilyGroup(DASolver.getOption("designSurfaceFamily"), DASolver.getOption("designSurfaces"))
  File "/home/lnhang/dafoam/packages/miniconda3/lib/python3.8/site-packages/dafoam/pyDAFoam.py", line 3743, in getOption
    raise Error("%s is not a valid option name." % name)
dafoam.pyDAFoam.Error
Finalising parallel run
Finalising parallel run
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[52961,1],1]
  Exit code:    1

**Reply by lnhang0618** [reply]
Here is the error raised while running runScript_v2.py in PeriodicHill_FieldInversion.

**Comment by friedenhe** [comment]
We recently updated the API and broke the v2 scripts. We just pushed an update to the tutorial repo and fixed the problem. The case should run without a problem now.

**Reply by lnhang0618** [reply]
Thanks a lot.The updated script can run now!

**Comment by lnhang0618** [comment]
Dear Professor He,

I am currently using the 3.0.5 version of DAFoam. I seem to have fallen into a loop in the "scripts_v2" of the Field Inversion (FI) module. I have to manually terminate it in the terminal, as it did not end as suggested in the tutorial with about 24 cores. Before I actively terminated it, it had already run over 800 objective functions. I also encountered an error when running this line in the runScript.py function: File "runScript.py", line 127, in <module> prob.setup(mode="rev").

Could you please advise me on how to identify the cause of this issue?

Best regards,
Luo Yinhang

**Reply by lnhang0618** [reply]
I will try running the program for a little longer.

**Comment by friedenhe** [comment]
Check opt_IPOPT.txt if this case converges. Running it for 800+ function evals should be pretty enough. Also, please run this case in serial in the future. We have an adjoint accuracy problem running this case in parallel. This issue is caused by the periodic boundary condition (we haven't fixed it yet). This is an issue for the PH case only. For other field inversion cases that do not have the periodic BC, you can run them in parallel with accurate adjoint gradients.

**Reply by lnhang0618** [reply]
Thank you for replying,the PH case can converge now.

**Comment by friedenhe** [comment]
BTW: the new runScript.py is for DAFoam-v3.0.6, which will be released soon.
